### ISSUE TITLE
Where conditions on related entities, case sensitivity issue on the alias in query

### ISSUE THREAD

**Issue by aymanfarhat** [Vinelab/NeoEloquent#49]
Hey guys, here is an issue I am facing when trying to use where to filter out a relation with another entity. 

For example:

I have the following relation:

``` php
public function investmentRounds()
{
     return $this->hasMany('InvestmentRound', 'RECEIVED');
}
```

``` php
$round = $startup->investmentRounds()
                 ->where('type', '=', $roundType)
                 ->get();
```

The respective query that gets generated is:

```
MATCH (startup:`Startup`:`Profile`), (startup)-[rel_received_investmentRounds:RECE
  IVED]->(investmentRounds:`InvestmentRound`) WHERE id(startup) = {idstartup} and inv
  estmentrounds.type = {type} and investmentrounds.type = {type_2} RETURN investmentRounds
```

Which obviously throws an exception, because the alias here is investmentRounds and not investmentrounds

```
SyntaxException: investmentrounds not defined (line 1, column 163)
```

So in summary, the alias for the related info is camel case just like the function defining this relationship, but for some reason it gets handled all as lower case in the rest of the query.

Probably the simplest way to just get going for now is to name that function all lowercase. As a quick workaround until the issue is solved I guess...

Thanks!

**Comment by JoshEllinger** [comment]
I can confirm this bug still exists.

**Comment by marlongichie** [comment]
Hi guys, I have also come across this bug recently.

Any updates?

**Comment by Mulkave** [comment]
Will try to steel some time tomorrow to work on this bug. @marlongichie and @JoshEllinger which version of NeoEloquent are you using ?

**Comment by marlongichie** [comment]
@Mulkave I am using: `"vinelab/neoeloquent": "1.1.*",`

**Comment by Mulkave** [comment]
@aymanfarhat @JoshEllinger @marlongichie calling you guys for reports on this issue when you get the time. Cheers 🍻

**Comment by marlongichie** [comment]
Updated to `v1.2.*` since coming across this issue and have not experienced the problem since. Haven't has a chance to see if it's also fixed in `v1.1.1`.
